src/StyleSheet.js

```diff
@@ -30,22 +29,26 @@ export default class StyleSheet {
     this.rules = Object.create(null)
     // Used to ensure correct rules order.
     this.rulesIndex = []
-    this.classes = Object.create(null)
+    this.classes = this.options.classes = Object.create(null)
```

Comment:
> What do you mean by all instances?

RulesContainer is used from sheet as well as from a ContainerRule

> flow on weak mode will not let you add additional properties

Good to know, lets tackle this step by step?